CODING_STANDARDS.md

```diff
@@ -0,0 +1,378 @@
+    
+_Example:_
+
+```tsx
+    import React, { PropsWithChildren, useEffect, useState } from 'react';
+    import { Platform } from 'react-native';
+    import Dayjs from 'dayjs';
+
+    import { useCreateChatContext } from './hooks/useCreateChatContext';
+    import { useIsOnline } from './hooks/useIsOnline';
+
+    import {
+        ChatContextValue,
+        ChatProvider,
+    } from '../../contexts/chatContext/ChatContext';
+    import { useOverlayContext } from '../../contexts/overlayContext/OverlayContext';
+    import {
+        DeepPartial,
+        ThemeProvider,
+    } from '../../contexts/themeContext/ThemeContext';
+    import {
+        TranslationContextValue,
+        TranslationProvider,
+    } from '../../contexts/translationContext/TranslationContext';
+    import { useStreami18n } from '../../utils/useStreami18n';
+
+    import type { Channel } from 'stream-chat';
+
+    import type { Theme } from '../../contexts/themeContext/utils/theme';
+    import type { Streami18n } from '../../utils/Streami18n';
+    import type {
+        DefaultAttachmentType,
+        DefaultChannelType,
+        DefaultCommandType,
+        DefaultEventType,
+        DefaultMessageType,
+        DefaultReactionType,
+        DefaultUserType,
+        UnknownType,
+    } from '../../types/types';
+```
+
+### __Destructure Props__
+
+If more than __1__ prop is being used within a method, it should be passed in as `props` and then destructured on the first line of the body.
+
+_Example single prop:_
+
+```tsx
+export const EmptyStateIndicator: React.FC<EmptyStateProps> = ({
+  listType,
+}) => {
+  ...
+};
+```
+
+_Example multiple props:_
+
+```tsx
+export const Avatar: React.FC<AvatarProps> = (props) => {
+  const {
+    containerStyle,
+    image: imageProp,
+    imageStyle,
+    name,
+    online,
+    presenceIndicator: presenceIndicatorProp,
+    presenceIndicatorContainerStyle,
+    size,
+    testID,
+  } = props;
+  ...
+};
+```
+
+### __Variable Declaration__
+
+Do not use `var`. Use `const` and avoid using `let` unless a creative solution cannot be implemented.
+
+Bad:
+
+```tsx
+let extra = attachment?.actions?.length ? 'actions' : '';
+if (
+    componentType === 'card' &&
+    !attachment.image_url &&
+    !attachment.thumb_url
+) {
+    extra = 'no-image';
+}
+```
+
+Good:
+
+```tsx
+const extra =
+    componentType === 'card' && !attachment?.image_url && !attachment?.thumb_url
+      ? 'no-image'
+      : attachment?.actions?.length
+      ? 'actions'
+      : '';
+```
+
+### __Magic Numbers__
+
+Avoid using magic numbers in code and instead create a searchable variable name for them
+
+Good:
+
+```tsx
+const DAY_IN_MILLISECONDS = 86400000;
+
+setTimeout(doSomething, DAY_IN_MILLISECONDS);
+```
+
+Bad:
+
+```tsx
+setTimeout(doSomething, 86400000);
+```
+
+### __Comments__
+
+Always use multiline comments instead of single line comments
+
+_Multiline Good:_
+
+```tsx
+/** Comment */
```

Comment:
I personally perceive `/** */` to be a doc comment, while `//` to be a code comment. Never thought about why though.

Reply:
haha same here.
eslint rule of max-len on comments will be useful I think.

Reply:
generally we should be coding in a way where variable and method names are self-documenting and so small code comments using `//` shouldn't be necessary. for explaining something in-depth the multiline comments are better than multiple lines of `//`

Reply:
It seems like an overkill to enfore `/** */`

Reply:
> so small code comments using // shouldn't be necessary

I don't find this necessary to be enforceable. I would rather use `//` for code comment and `/** */` for js doc or comments which is big enough. We should leave this for dev to decide.

Reply:
From w3: `Comment when there is an important thing to say, and if you do comment use the /* */ notation. Single line comments using // can be problematic if people minify your code without stripping comments and in general are less versatile.`

I personally prefer multi because it enforces some spacing to add readability; I feel like I come across things that should be multiline done as multiple lines of `//` more than being bothered by `/** Comment */`. I was trying to find this guide I looked at years ago, but didn't spot it, had a whole bunch about spacing out of comments (always empty newline above) etc. It was that guide that had me switch to only multiline (except ignores).

Reply:
We can go with multiline comment, for the reason of personal preference.

> Single line comments using // can be problematic if people minify your code without stripping comments and in general are less versatile.

But I just don't count this as a reason. This doc is super ancient, if you look at its history. And origin of that doc is [long dead](https://dev.opera.com/blog/the-web-standards-curriculum-is-dead-long-live-webplatform-org/).  I would be amazed if there is any uglifier/minifier out there which doesn't remove `//` as part of stripping comments. Its a standard javascript syntax.

Reply:
O it's def a standard syntax, just found that w3 stuff while searching for this old guide I read; looked again still no luck. Good spacing, readability, and consistency is my push. stream-chat-js with it's HUGE variety of comments (multi, single, inline, JSDocs, etc.) feels like it has had too many cooks in the kitchen. Some code is over commented, while elsewhere there are none, and the spacing is not consistent at all; I actually find the comments distracting in that repo personally.

Reply:
Makes sense!! I am fine with `/** */` !!